platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/EmbeddedSolrFiles.java

```diff
@@ -0,0 +1,258 @@
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.solr.factory.impl;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.io.Closeables;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Objects;
+import javax.annotation.Nullable;
+import javax.xml.parsers.ParserConfigurationException;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.Validate;
+import org.apache.solr.core.SolrConfig;
+import org.apache.solr.schema.IndexSchema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+/**
+ * Maintain embedded Solr configuration and schema files.
+ *
+ * <p><i>Note:</i> The corresponding {@link SolrConfig} and {@link IndexSchema} will be constructed
+ * the first time they are retrieved.
+ */
+class EmbeddedSolrFiles {
+  private static final Logger LOGGER = LoggerFactory.getLogger(EmbeddedSolrFiles.class);
+
+  private final String coreName;
+  private final String configName;
+  private final File configFile;
+  private final String schemaName;
+  private final File schemaFile;
+  private final ConfigurationFileProxy configProxy;
+  private volatile SolrConfig solrConfig = null;
+  private volatile IndexSchema schemaIndex = null;
+  private volatile String dataDirPath = null;
+
+  @VisibleForTesting // required by Spock to spy on an existing object
+  EmbeddedSolrFiles() {
+    this.coreName = null;
+    this.configName = null;
+    this.configFile = null;
+    this.schemaName = null;
+    this.schemaFile = null;
+    this.configProxy = null;
+  }
+
+  /**
+   * Constructor.
+   *
+   * @param coreName name of the Solr core
+   * @param configXml name of the Solr configuration file
+   * @param schemaXmls file names of the Solr core schemas to attempt to load (will start with the
+   *     first and fallback to the others in order if unavailable)
+   * @param configProxy {@link ConfigurationFileProxy} instance to use
+   * @throws IllegalArgumentException if <code>coreName</code>, <code>configXml</code>, <code>
+   *     schemaXmls</code>, or <code>configProxy</code> is <code>null</code> or if unable to find
+   *     any files
+   */
+  public EmbeddedSolrFiles(
+      String coreName, String configXml, String[] schemaXmls, ConfigurationFileProxy configProxy) {
+    Validate.notNull(coreName, "invalid null Solr core name");
```

Comment:
All this code was moved around to reduce cyclomatic complexity and since it will be going away with the Solr as an external process story, I will keep it as is for now.